rbd/rbd.go

```diff
@@ -1101,6 +1115,13 @@ func OpenImageReadOnly(ioctx *rados.IOContext, name, snapName string) (*Image, e
 //  int rbd_open_by_id(rados_ioctx_t io, const char *id,
 //                     rbd_image_t *image, const char *snap_name);
 func OpenImageById(ioctx *rados.IOContext, id, snapName string) (*Image, error) {
+	if ioctx == nil {
```

Comment:
can you avoid code duplication by creating a new helper/validator function?

Reply:
Unfortunately I do not see how. We have `Image.validate()`, but we can not use it in this case where different parameters are passed.

Reply:
It's unfortunate but I agree with @nixpanic there's not much else to do for these.